src/main/java/org/openml/webapplication/features/GlobalMetafeatures.java

```diff
@@ -23,36 +23,59 @@
     private int expectedQualities;
     private ArrayList<String> expectedIds = new ArrayList<>();
 
-    public GlobalMetafeatures(Integer window_size) throws Exception {
-        Characterizer[] characterizers = {
-                new SimpleMetaFeatures(), // done before, but necessary for streams
-                new Statistical(),
-                new NominalAttDistinctValues(),
-                new AttributeEntropy(),
-                new GenericLandmarker("kNN1N", cp1NN, 2, null),
-                new GenericLandmarker("NaiveBayes", cpNB, 2, null),
-                new GenericLandmarker("DecisionStump", cpDS, 2, null),
-                new GenericLandmarker("CfsSubsetEval_kNN1N", cpASC, 2, Utils.splitOptions(preprocessingPrefix + cp1NN)),
-                new GenericLandmarker("CfsSubsetEval_NaiveBayes", cpASC, 2, Utils.splitOptions(preprocessingPrefix + cpNB)),
-                new GenericLandmarker("CfsSubsetEval_DecisionStump", cpASC, 2, Utils.splitOptions(preprocessingPrefix + cpDS))
-        };
-        batchCharacterizers = new ArrayList<>(Arrays.asList(characterizers));
-        // additional parameterized batch landmarkers
-        String zeros = "0";
-        for( int i = 1; i <= 3; ++i ) {
-            zeros += "0";
-            String[] j48Option = { "-C", "." + zeros + "1" };
-            batchCharacterizers.add(new GenericLandmarker("J48." + zeros + "1.", "weka.classifiers.trees.J48", 2, j48Option));
+    /**
+     * Collection of characterizers
+     * 
+     * @param characterizer Class name of requested characterizer. Can also be 'fast' 
+     * for everything except landmarkers. If null, all characterizers are added.
+     * @param window_size for streaming landmarkers
+     * @throws Exception
+     */
+    public GlobalMetafeatures(String characterizer, Integer window_size) throws Exception {
+    	
+    	ArrayList<Characterizer> batchCharacterizers = new ArrayList<Characterizer>();
```

Comment:
Shouldn't we make ArrayList<Characterizer> batchCharacterizers a TreeMap? Mapping from String (name) to Characterizer. We could then specify with a string keyword a single characterizer that should be invoked.

Reply:
I'm not sure, we just convert that to a list of IDs anyway? Do we invoke them by name anywhere? Also, landmarkers have names, other characterizers not?